pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -2647,3 +2663,167 @@ func handleDNSDelete(ctxArg interface{}, key string,
 	maybeHandleDNS(ctx)
 	log.Infof("handleDNSDelete done for %s\n", key)
 }
+
+func validateAppNetworkConfig(ctx *zedrouterContext, appNetConfig types.AppNetworkConfig,
+	appNetStatus *types.AppNetworkStatus) bool {
+	log.Infof("AppNetwork(%s), check for duplicate port map acls", appNetConfig.DisplayName)
+	// For App Networks, check for common port map rules
+	ulCfgList0 := appNetConfig.UnderlayNetworkList
+	if len(ulCfgList0) == 0 {
+		return false
+	}
+	sub := ctx.subAppNetworkConfig
+	items := sub.GetAll()
+	for _, cfg := range items {
+		appNetConfig1 := cast.CastAppNetworkConfig(cfg)
+		ulCfgList1 := appNetConfig1.UnderlayNetworkList
+		if appNetConfig.DisplayName == appNetConfig1.DisplayName ||
+			len(ulCfgList1) == 0 {
+			continue
+		}
+		ret, err := checkUnderlayNetworkForDuplicatePortMap(ctx, ulCfgList0,
+			ulCfgList1)
+		if err != nil {
+			addError(ctx, appNetStatus, "portmap for underlay", err)
+			return true
+		}
+		if ret {
+			log.Errorf("app Network(%s) have overlapping portmap rule with %s\n",
+				appNetStatus.DisplayName, appNetConfig1.DisplayName)
+			errStr := fmt.Sprintf("portmap rule overlap with %s",
+				appNetConfig1.DisplayName)
+			err := errors.New(errStr)
+			addError(ctx, appNetStatus, "portmap for underlay", err)
+			return true
+		}
+	}
+	return false
+}
+
+func checkUnderlayNetworkForDuplicatePortMap(ctx *zedrouterContext,
+	ulCfgList0 []types.UnderlayNetworkConfig, ulCfgList1 []types.UnderlayNetworkConfig) (bool, error) {
+	for _, ulCfg0 := range ulCfgList0 {
+		network0 := ulCfg0.Network.String()
+		for _, ulCfg1 := range ulCfgList1 {
+			network1 := ulCfg1.Network.String()
+			match, err := networkInstancesWithUplinkOverlap(ctx, network0, network1)
+			if err != nil {
+				return false, err
+			}
+			if match {
+				if matchACLsForPortMap(ulCfg0.ACLs, ulCfg1.ACLs) {
+					return true, nil
+				}
+			}
+		}
+	}
+	return false, nil
+}
+
+// network instance or, sharing common uplink (ephimeral or, physical port)
+func networkInstancesWithUplinkOverlap(ctx *zedrouterContext, network string,
+	network1 string) (bool, error) {
+	if network == network1 {
+		log.Debugf("Same Network Instance, %s\n", network)
+		return true, nil
+	}
+	netInstConfig := lookupNetworkInstanceConfig(ctx, network)
```

Comment:
I think it would be both simpler and more robust to future changes to look at NetworkInstanceStatus and just look for overlap in the IfNameList and not have to worry about the special labels like uplink and freeuplink. Me might add more arbitrary labels down the road.

Reply:
sure. done